inspirehep/modules/multieditor/actions.py

```diff
@@ -0,0 +1,209 @@
+        if not record[key]:
+            del record[key]
+
+
+def create_schema_record(schema, path, value):
+    """Object creation in par with the schema."""
+    record = {}
+    temp_record = record
+    new_schema = schema
+    if new_schema['type'] == 'array':
+        new_schema = new_schema['items']['properties']
+    elif new_schema['type'] == 'object':
+        new_schema = new_schema['properties']
+    for key in path:
+        new_schema = new_schema[key]
+        if new_schema['type'] == 'object':
+            new_schema = schema['properties']
+            temp_record[key] = {}
+            temp_record = temp_record[key]
+
+        elif new_schema['type'] == 'array':
+            if new_schema['items']['type'] == 'object':
+                new_schema = new_schema['items']['properties']
+                temp_record[key] = [{}]
+                temp_record = temp_record[key][0]
+    temp_record[path[-1]] = value
+    return record
+
+
+def check_value(record, keys, value_to_check):
+    """Where continues to find the value."""
+    new_keys = deque(keys)
+    key = new_keys.popleft()
+    if key not in record:
+        return False
+    temp_record = record[key]
+    if isinstance(temp_record, list):
+        for index, array_record in enumerate(temp_record):
+            if len(new_keys) == 0:
+                if array_record == value_to_check:
+                    return True
+            else:
+                if check_value(array_record,
+                               new_keys, value_to_check):
+                    return True
+    else:
+        if len(new_keys) == 0:
+            if temp_record == value_to_check:
+                return True
+        else:
+            return check_value(temp_record,
+                               new_keys, value_to_check)
+    return False
+
+
+def apply_action_to_field(record, key, regex,
+                          action, value, value_to_check):
+    """Function for applying action to object or array."""
+    if isinstance(record[key], list):
+        apply_to_array(record, key, regex, action, value, value_to_check)
+    else:
+        apply_to_object(record, key, regex, action, value, value_to_check)
+
+
+def apply_to_array(record, key, regex, action, value, values_to_check):
+    """Applying action to array."""
+    for i in range(len(record[key])):
+        if action == 'Update':
+            if regex and re.search(
+                    re.escape(values_to_check[0]),
+                    record[key][i]):
+                record[key][i] = value
+            elif record[key][i] in values_to_check:
+                record[key][i] = value
+        elif action == 'Addition':
+            record[key].append(value)
+            return  # In that case we want to stop looping
+        elif action == 'Deletion':
+            if len(values_to_check) == 0 \
+                    or record[key][i] in values_to_check:
+                record[key].pop(i)
+                --i
```

Comment:
This shows that there is no test covering this part, since otherwise you would catch that `--i` does not have any effect ;)

Reply:
Yup got it! I updated a test to get this case as well and I did some fixing on thsi code.